tests/unit/lib/test_datachain.py

```diff
@@ -3350,3 +3350,108 @@ def test_compare_right_compare_wrong_length(test_session):
     assert str(exc_info.value) == (
         "'compare' and 'right_compare' must be have the same length"
     )
+
+
+@pytest.mark.parametrize("added", (True, False))
+@pytest.mark.parametrize("deleted", (True, False))
+@pytest.mark.parametrize("modified", (True, False))
+@pytest.mark.parametrize("unchanged", (True, False))
+@pytest.mark.parametrize("status_col", ("diff", None))
```

Comment:
The test looks overcomplicated 🙂

Is it possible to separate this to small tests instead of large parametrized one?

Also, I don't see value in all modified, deleted, unchanged statuses since compare() tests have to cover this, not diff(). The same for `status_col`. 2-3 simple tests should cover diff() pretty well.

Reply:
Simplified it